azure/services/bastionhosts/azurebastion.go

```diff
@@ -69,7 +79,7 @@ func (s *Service) ensureAzureBastion(ctx context.Context, azureBastionSpec azure
 						Name: to.StringPtr(bastionHostIPConfigName),
 						BastionHostIPConfigurationPropertiesFormat: &network.BastionHostIPConfigurationPropertiesFormat{
 							Subnet: &network.SubResource{
-								ID: subnet.ID,
+								ID: existingSubnet.ID,
```

Comment:
if just using the subnet to grab it's ID we can use the `SubnetID` helper function to construct the ID from the sub ID, rg, vnet name and subnet name and get rid of that extra GET call